sdks/python/apache_beam/io/avroio_it_test.py

```diff
@@ -0,0 +1,237 @@
+  }
+
+
+class AvroITTestBase(object):
+
+  def setUp(self):
+    self.SCHEMA_STRING = '''
+      {"namespace": "example.avro",
+       "type": "record",
+       "name": "User",
+       "fields": [
+           {"name": "label", "type": "string"},
+           {"name": "number",  "type": ["int", "null"]},
+           {"name": "number_str", "type": ["string", "null"]},
+           {"name": "color", "type": ["string", "null"]}
+       ]
+      }
+      '''
+    self.uuid = str(uuid.uuid4())
+
+  @attr('IT')
+  def avro_it_test(self):
+    write_pipeline = TestPipeline(is_integration_test=True)
+    read_pipeline = TestPipeline(is_integration_test=True)
+    output = '/'.join([write_pipeline.get_option('output'),
+                       self.uuid, 'avro-files'])
+    files_output = '/'.join([output, 'avro'])
+
+    num_records = write_pipeline.get_option('records')
+    num_records = int(num_records) if num_records else 10000
+
+    # Seed a `PCollection` with indices that will each be FlatMap'd into
+    # `batch_size` records, to avoid having a too-large list in memory at
+    # the outset
+    batch_size = write_pipeline.get_option('batch-size')
+    batch_size = int(batch_size) if batch_size else 1000
+
+    assert batch_size < num_records
+    # pylint: disable=range-builtin-not-iterating
+    batches = range(int(num_records / batch_size))
+
+    def batch_indices(start):
+      # pylint: disable=range-builtin-not-iterating
+      return range(start * batch_size, (start + 1) * batch_size)
+
+    multiplier = 10
+    assert (multiplier % 2 == 0), \
+      "Use an even multiplier to not compare floats."
+
+    # A `PCollection` with `num_records` avro records
+    _ = (write_pipeline
+         | 'CreateBatches' >> Create(batches)
+         | 'ExpandBatches' >> FlatMap(batch_indices)
+         | 'Multiply' >> Map(lambda x: multiplier * x)
+         | 'CreateRecords' >> Map(record)
+         | 'WriteAvro' >> WriteToAvro(files_output,
+                                      self.SCHEMA,
+                                      file_name_suffix=".avro",
+                                      use_fastavro=self.use_fastavro)
+        )
+
+    write_result = write_pipeline.run()
+    write_result.wait_until_finish()
+    assert write_result.state == PipelineState.DONE
+
+    def check(x, mean):
+      if not x == mean:
+        raise BeamAssertException('Assertion failed: %s == %s' % (x, mean))
+
+    def check_distr(lst):
+      if not len(set(lst)) == 1:
+        raise BeamAssertException(
+            'Labels should be equally distributed: %s ' % (lst))
+
+    def check_concat(concatenated_string):
+      split_concatenated_string = concatenated_string.split(" ")
+      for i in range(len(batches) * batch_size):
+        if str(multiplier * i) not in split_concatenated_string:
+          raise BeamAssertException(
+              'String should be in concatenation: %s' % str(multiplier * i))
+
+    def check_for_none(lst):
+      if not None in lst:
```

Comment:
Old comment valentyn: We could simplify to 
`if None not in list:
  raise ...`
Resolution: simplified